entity-service-impl/src/main/java/org/hypertrace/entity/query/service/EntityQueryServiceImpl.java

```diff
@@ -543,9 +555,78 @@ public void total(
     responseObserver.onCompleted();
   }
 
+  @Override
+  public void deleteEntities(
+      DeleteEntitiesRequest request, StreamObserver<DeleteEntitiesResponse> responseObserver) {
+    RequestContext requestContext = RequestContext.CURRENT.get();
+    if (validateDeleteEntitiesRequest(request, requestContext)) {
+      responseObserver.onError(new ServiceException("Invalid request provided"));
+      return;
+    }
+
+    List<String> entityIds = getEntityIdsToDelete(requestContext, request);
+    if (entityIds.size() == 0) {
+      LOG.info("{}. No entities found to delete", request);
+      responseObserver.onNext(DeleteEntitiesResponse.newBuilder().build());
+      responseObserver.onCompleted();
+      return;
+    }
+
+    if (entityIds.size() > maxEntitiesToDelete) {
+      LOG.info(
+          "{}. Number of ids to delete exceeds the maximum limit: Entity Ids limit exceeded",
+          request);
+      responseObserver.onError(new RuntimeException("Entity Ids limit exceeded"));
+      return;
+    }
+
+    if (LOG.isDebugEnabled()) {
+      LOG.debug("Deleting entity of type: {} and ids: {}", request.getEntityType(), entityIds);
+    }
+
+    Optional<String> tenantId = requestContext.getTenantId();
+    this.entitiesCollection.delete(
```

Comment:
Always wrap the whole thing in a try/catch and report the caught error. If this errors out, currently it would result in a hanging request rather than reporting that back to the caller.